core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -301,6 +301,10 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
           // enqueue for retry
           txnLogAppendRetryQueue.add(txnLogAppend)
 
+        case Errors.COORDINATOR_LOAD_IN_PROGRESS =>
```

Comment:
Yes that is possible. In this case the markers would be written twice, but that would be fine. Here is my reasoning:

0). Thread 1 set the loading partition in the set, and start loading the partition;

1). Thread 2 complete sending the markers and appending to the log, and now tries to complete the transition; but `getMetadata` returns `loading in progress`, and then abort completing the append. Now there is a race condition and either of the following may happen

2.1) Thread 1 may not see the appended `completeXX` txn log from thread 2, and then tries to complete the sending markers and re-append. It will result in duplicate markers and duplicate `completeXX` log.

2.2) Thread 2 may have seen the appended `completeXX` txn log from thread 2, and then do nothing.